### DISCUSSION TITLE
Back-relation. I need some feature extension

### DISCUSSION THREAD

**Discussion by sandbox-apps** [prisma/prisma#8850] [Q&A] [answered]
Here is my models with its corresponding relationships to one another.

```prisma
model User {
  id          String @id @default(cuid())
  
  memberships Membership[]
}

model Group {
  id          String @id @default(cuid())
  name        String
  
  memberships Membership[]
}

model Membership {
  groupId String
  group   Group @relation(fields: [groupId], references: [id])
  id      String
  user    User @relation(fields: [id], references: [id])

  pending Boolean

  @@id([groupId, id])
}
```

I want to find all groups with my corresponding membership status.
For example:
1. If membership row not found, it means `include query` returned null
2. If membership found, we can check if status is `pending` or not

Problem
1. `Membership` field for `group` return an array because of its back relation

Solution
1. We must be able to define a singular field for `memberships`

Syntax (Problem)
```ts
const me = {id: 'user-id'};

this.prisma.group.findMany({
  include: {
    memberships: {
      where: {
        groupId: 'you can`t get the group ids from the find many',
        id: me.id,
      },
    },
  },
});
```

Expected Available Syntax (Solution)
```ts
const me = {id: 'user-id'};

this.prisma.group.findMany({
  include: {
    membership: {
      where: {
        id: me.id,
      },
    },
  },
});
```

Notice that I removed the `groupId` in the where clause. So it must infer the `groupId` from the result of `findMany`

Expected Output
```json
[
  {
   "id": "1",
   "name": "Group 1",
   "membership": null,
  },
  {
   "id": "2",
   "name": "Group 2",
   "membership": {
     "groupId": "2",
     "id": "1",
     "pending": true,
    },
  },
  {
   "id": "2",
   "name": "Group 3",
   "membership": {
     "groupId": "2",
     "id": "1",
     "pending": false,
    },
  },
],
```

Any idea how to achieve this in prisma?
I get this idea from mongoose. Before I came to prisma. I'm a heavily (MongoDB - mongoose user)

https://mongoosejs.com/docs/populate.html
https://mongoosejs.com/docs/tutorials/virtuals.html#populate

**Comment by ryands17** [comment, marked as answer]
@sandbox-apps 👋 

I would suggest creating a feature request for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that we can look into this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
